# Task-1 Making sense of the data

One of the reporters asks if you can find Mahmoud Khalil in the dataset. If you do, explain how you identified the correct record in the ICE data, by describing what logic you used, which fields you filtered on, how you handled any ambiguities and what steps you took to be confident it was the right match. 

## 0. Set up

In [1]:
from pathlib import Path
from datetime import datetime

import numpy as np
import pandas as pd

import process_data

In [2]:
arrests_filename = 'arrests-0923-0625.xlsx'
cwd = Path.cwd()
root = cwd.parent
data = root / "data"

In [32]:
pd.set_option("display.max_rows", 300)

In [3]:
arrests_df = process_data.read_arrests_data(data/arrests_filename)

In [4]:
arrests_df.head()

apprehension_date apprehension_state                   apprehension_aor  \
0 2024-08-07 09:43:00           VIRGINIA  WASHINGTON AREA OF RESPONSIBILITY   
1 2024-10-19 20:33:00              TEXAS     HOUSTON AREA OF RESPONSIBILITY   
2 2025-04-15 10:08:21         NEW JERSEY      NEWARK AREA OF RESPONSIBILITY   
3 2025-06-03 09:20:00          MINNESOTA    ST. PAUL AREA OF RESPONSIBILITY   
4 2025-01-21 17:41:00                NaN       MIAMI AREA OF RESPONSIBILITY   

                final_program        apprehension_method  \
0  ERO CRIMINAL ALIEN PROGRAM       NON-CUSTODIAL ARREST   
1  ERO CRIMINAL ALIEN PROGRAM    CAP LOCAL INCARCERATION   
2  ERO CRIMINAL ALIEN PROGRAM  CAP FEDERAL INCARCERATION   
3         FUGITIVE OPERATIONS       NON-CUSTODIAL ARREST   
4  ERO CRIMINAL ALIEN PROGRAM    CAP LOCAL INCARCERATION   

       apprehension_criminality                           case_status  \
0          1 CONVICTED CRIMINAL  8-EXCLUDED/REMOVED - INADMISSIBILITY   
1          1 CONVICTED CRIMINAL    6-DEPORTED/REMOVED - DEPORTABILITY   
2          1 CONVICTED CRIMINAL  8-EXCLUDED/REMOVED - INADMISSIBILITY   
3  3 OTHER IMMIGRATION VIOLATOR                                ACTIVE   
4    2 PENDING CRIMINAL CHARGES       3-VOLUNTARY DEPARTURE CONFIRMED   

                                       case_category departed_date  \
0                        [16] REINSTATED FINAL ORDER    2024-08-19   
1                        [16] REINSTATED FINAL ORDER    2024-10-22   
2                        [16] REINSTATED FINAL ORDER    2025-06-10   
3    [8G] EXPEDITED REMOVAL - CREDIBLE FEAR REFERRAL           NaT   
4  [8C] EXCLUDABLE / INADMISSIBLE - ADMINISTRATIV...    2025-02-01   

    departure_country final_order_yes_no  birth_year citizenship_country  \
0            HONDURAS                YES        1981            HONDURAS   
1              MEXICO                YES        1984              MEXICO   
2  DOMINICAN REPUBLIC                YES        1988  DOMINICAN REPUBLIC   
3                 NaN                YES        1985            COLOMBIA   
4              MEXICO                YES        1983              MEXICO   

   gender                         apprehension_site_landmark  \
0    MALE                     HBG GENERAL AREA, NON-SPECIFIC   
1    MALE                    HARRIS COUNTY JAIL, HOUSTON, TX   
2    MALE                          FORT DIX EAST, NEW JERSEY   
3  FEMALE                     SPM GENERAL AREA, NON-SPECIFIC   
4    MALE  MIAMI DADE COUNTY JAIL TURNER GUILFORD KNIGHT ...   

                          unique_identifier  
0  0000b34edd657d516c02b13a7c352d62d0effcb6  
1  0000ba6e459998a6046d185d82cf4349de1479d0  
2  0000c3d23fb0e444864559575900d410c4e8490f  
3  0000d3dbf8033b5f209f6547ffee5b84feb4f599  
4  000104d730bf021326c6dc0deb3dd575304136b5

# Approach:

**Note:** for initial data exploration please see `./0-data_exploration.ipynb`

## 1. Choosing what to filter on:

**What information do we know about Mahmoud Khalil that we can use to find him in the dataset?**

Things that have been reported about Mahmoud Khalil:
* Location of arrest: New York (Columbia University Residence)
* Year of birth: 1995
* Citizenship: Algeria (born in Syria to Palestinian parents, and moved to Lebanon)
* He had a green card (ie not on a tempory visa or undocumented)
* Time and Date of Arrest: 2025-03-08, around 8.30pm 
* Where was he taken after his arrest? [his own understanding of this was shared in his letter](https://www.newsweek.com/mahmoud-khalil-columbia-hamas-gaza-israel-letter-2047002)
* * 26 Federal Plaza - on the 8th March (?)
  * Elizabeth Detention Centre, New Jersey - taken in the early hours of the 9th (NB when his wife tried to find him there on the 9th she was told he was not there, so there are some question marks over this)
  * ICE detention center in Louisiana 



### Finding the relevant variables:

In [7]:
birth_1995 = arrests_df['birth_year'] == 1995
citizenship_algeria = arrests_df['citizenship_country'] == 'ALGERIA' # might need to try out SYRIA and LEBANON here (although I don't think he has citizenship)

In [8]:
unique_vars_filter_str = lambda col, str_match: arrests_df[col][arrests_df[col].fillna('').str.contains(str_match)].unique()

#### Locations

##### 1. Arrest location = New York, initially taken to 26 Federal Plaza

In [7]:
np.sort(arrests_df['apprehension_state'].fillna('').unique())

array(['', 'ALABAMA', 'ALASKA', 'ARIZONA', 'ARKANSAS',
       'ARMED FORCES - EUROPE', 'ARMED FORCES - THE AMERICAS',
       'ARMED SERVICES - PACIFIC', 'CALIFORNIA', 'COLORADO',
       'CONNECTICUT', 'DELAWARE', 'DISTRICT OF COLUMBIA',
       'FEDERATED STATES OF MICRONESIA', 'FLORIDA', 'GEORGIA', 'GUAM',
       'HAWAII', 'IDAHO', 'ILLINOIS', 'INDIANA', 'IOWA', 'KANSAS',
       'KENTUCKY', 'LOUISIANA', 'MAINE', 'MARYLAND', 'MASSACHUSETTS',
       'MEXICO', 'MICHIGAN', 'MINNESOTA', 'MISSISSIPPI', 'MISSOURI',
       'MONTANA', 'NEBRASKA', 'NEVADA', 'NEW HAMPSHIRE', 'NEW JERSEY',
       'NEW MEXICO', 'NEW YORK', 'NORTH CAROLINA', 'NORTH DAKOTA',
       'NORTHERN MARIANA ISLANDS', 'OHIO', 'OKLAHOMA', 'OREGON',
       'PENNSYLVANIA', 'PUERTO RICO', 'RHODE ISLAND', 'SOUTH CAROLINA',
       'SOUTH DAKOTA', 'TAMAULIPAS', 'TENNESSEE', 'TEXAS', 'UTAH',
       'VERMONT', 'VIRGIN ISLANDS', 'VIRGINIA', 'WASHINGTON',
       'WEST VIRGINIA', 'WISCONSIN', 'WYOMING'], dtype=object)

In [8]:
unique_vars_filter_str('apprehension_site_landmark','26 FEDERAL')

array(['NDD - 26 FEDERAL PLAZA NY, NY'], dtype=object)

In [9]:
arrests_df[arrests_df['apprehension_site_landmark']=='NDD - 26 FEDERAL PLAZA NY, NY']['apprehension_state'].value_counts(dropna=False)

apprehension_state
NEW YORK          5646
NaN                151
NORTH CAROLINA       4
NEW JERSEY           1
FLORIDA              1
Name: count, dtype: int64



**Observations:**
* Looks like that is the correct site for 26 Federal Plaza, NY
* But note - not all who have `'apprehension_site_landmark' == 'NDD - 26 FEDERAL PLAZA NY, NY'` have `'apprehension_state' == 'NEW YORK'`
   * Maybe taken to be processed at 26 Federal Plaza after being initially arrested in the location? But Florida and North Carolina are quite a long way
   * 151 with missing `apprehension_state`

In [24]:
state_ny = arrests_df['apprehension_state'] == 'NEW YORK' 
state_missing = arrests_df['apprehension_state'].isna() # NB - from initial data exploration we know this has quite a lot of missing, so don't rely on it being added
loc_federal_plaza = arrests_df['apprehension_site_landmark'] == 'NDD - 26 FEDERAL PLAZA NY, NY'

##### 2. Taken to: Elizabeth Detention Center in New Jersey and ICE Detention Center in Louisiana

Doesn't look like these are in this dataset, which makes sense because this is arrests, not detentions:

In [10]:
unique_vars_filter_str('apprehension_site_landmark','ELIZABETH')

array(['ELIZABETH CITY', 'ELIZABETH PD'], dtype=object)

In [11]:
unique_vars_filter_str('apprehension_site_landmark','LOUISIANA')

array(['LOUISIANA STATE PENITENTIARY - ANGOLA'], dtype=object)

#### 2. Apprehension date

In [11]:
get_date = lambda date_str: datetime.strptime(date_str, '%Y-%m-%d').date()

In [12]:
def filter_dates(df, start_date, eq=True, end_date=None, date_col='apprehension_date'):
    if end_date:
        return (df[date_col].dt.date >= get_date(start_date)) & (df[date_col].dt.date <= get_date(end_date))
    elif eq:
        return df[date_col].dt.date == get_date(start_date)
    else:
        return df[date_col].dt.date >= get_date(start_date)

In [13]:
arrests_df[filter_dates(arrests_df, '2025-03-08')]

apprehension_date apprehension_state  \
429    2025-03-08 22:05:00              TEXAS   
1801   2025-03-08 08:53:00            FLORIDA   
4209   2025-03-08 23:36:00            GEORGIA   
6619   2025-03-08 09:47:00         WASHINGTON   
6848   2025-03-08 07:13:00              TEXAS   
...                    ...                ...   
264332 2025-03-08 04:39:00              TEXAS   
264333 2025-03-08 09:50:11              TEXAS   
264334 2025-03-08 07:54:37              TEXAS   
264335 2025-03-08 09:47:51              TEXAS   
264336 2025-03-08 09:51:28              TEXAS   

                        apprehension_aor                final_program  \
429       HOUSTON AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
1801        MIAMI AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
4209      ATLANTA AREA OF RESPONSIBILITY                 287G PROGRAM   
6619      SEATTLE AREA OF RESPONSIBILITY  NON-DETAINED DOCKET CONTROL   
6848    HARLINGEN AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
...                                  ...                          ...   
264332     DALLAS AREA OF RESPONSIBILITY          FUGITIVE OPERATIONS   
264333  HARLINGEN AREA OF RESPONSIBILITY  NON-DETAINED DOCKET CONTROL   
264334  HARLINGEN AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   
264335  HARLINGEN AREA OF RESPONSIBILITY      DETAINED DOCKET CONTROL   
264336  HARLINGEN AREA OF RESPONSIBILITY   ERO CRIMINAL ALIEN PROGRAM   

              apprehension_method      apprehension_criminality  \
429       CAP LOCAL INCARCERATION    2 PENDING CRIMINAL CHARGES   
1801                      LOCATED  3 OTHER IMMIGRATION VIOLATOR   
4209               287(G) PROGRAM    2 PENDING CRIMINAL CHARGES   
6619                      LOCATED          1 CONVICTED CRIMINAL   
6848    CAP FEDERAL INCARCERATION  3 OTHER IMMIGRATION VIOLATOR   
...                           ...                           ...   
264332       NON-CUSTODIAL ARREST  3 OTHER IMMIGRATION VIOLATOR   
264333    CAP LOCAL INCARCERATION  3 OTHER IMMIGRATION VIOLATOR   
264334    CAP LOCAL INCARCERATION    2 PENDING CRIMINAL CHARGES   
264335                    LOCATED    2 PENDING CRIMINAL CHARGES   
264336    CAP LOCAL INCARCERATION    2 PENDING CRIMINAL CHARGES   

                                 case_status  \
429     8-EXCLUDED/REMOVED - INADMISSIBILITY   
1801    8-EXCLUDED/REMOVED - INADMISSIBILITY   
4209    8-EXCLUDED/REMOVED - INADMISSIBILITY   
6619      6-DEPORTED/REMOVED - DEPORTABILITY   
6848      6-DEPORTED/REMOVED - DEPORTABILITY   
...                                      ...   
264332                                   NaN   
264333                        9-VR WITNESSED   
264334                        9-VR WITNESSED   
264335                        9-VR WITNESSED   
264336                        9-VR WITNESSED   

                                            case_category departed_date  \
429     [8C] EXCLUDABLE / INADMISSIBLE - ADMINISTRATIV...    2025-05-03   
1801                          [16] REINSTATED FINAL ORDER    2025-03-18   
4209    [8C] EXCLUDABLE / INADMISSIBLE - ADMINISTRATIV...    2025-05-12   
6619                          [16] REINSTATED FINAL ORDER    2025-06-03   
6848                          [16] REINSTATED FINAL ORDER    2025-03-12   
...                                                   ...           ...   
264332                                                NaN           NaT   
264333                            [9] VR UNDER SAFEGUARDS    2025-03-09   
264334                            [9] VR UNDER SAFEGUARDS    2025-03-08   
264335                            [9] VR UNDER SAFEGUARDS    2025-03-08   
264336                            [9] VR UNDER SAFEGUARDS    2025-03-08   

       departure_country final_order_yes_no  birth_year citizenship_country  \
429            GUATEMALA                YES        2002           GUATEMALA   
1801           GUATEMALA                YES        1988           GUATEMALA   
4209           NICARAGU

#### 3. Immigration status

According to the documentation, Unique Identifier is based on Alien Registration Number (A-number). As Mahmoud Khalil had a green card, we might assume he would not have an A-number, and therefore wouldn't have a Unique Identifier. This is a big assumption though, and the ICE enforcement agents did think he had a student visa, so he might have an A-number from before he had a green card?

In [64]:
no_ui = arrests_df['unique_identifier'].isna()

Are there any other clues in other variables that indicate immigration status?

In [65]:
arrests_df['final_program'].value_counts()

final_program
ERO CRIMINAL ALIEN PROGRAM             162433
FUGITIVE OPERATIONS                     39776
NON-DETAINED DOCKET CONTROL             33165
ALTERNATIVES TO DETENTION               12761
DETAINED DOCKET CONTROL                  8972
287G PROGRAM                             4934
MOBILE CRIMINAL ALIEN TEAM               2214
JUVENILE                                  444
ERO CRIMINAL PROSECUTIONS                 304
DETENTION AND DEPORTATION                 159
LAW ENFORCEMENT AREA RESPONSE UNITS        45
287G TASK FORCE                            10
VIOLENT CRIMINAL ALIEN SECTION              5
JOINT CRIMINAL ALIEN RESPONSE TEAM          4
Name: count, dtype: int64

In [66]:
arrests_df['apprehension_method'].value_counts()

apprehension_method
CAP LOCAL INCARCERATION                        112103
NON-CUSTODIAL ARREST                            56910
LOCATED                                         31405
CAP FEDERAL INCARCERATION                       23408
CAP STATE INCARCERATION                         10419
OTHER EFFORTS                                    9018
ERO REPROCESSED ARREST                           8875
287(G) PROGRAM                                   6313
PROBATION AND PAROLE                             3833
LAW ENFORCEMENT AGENCY RESPONSE UNIT              845
OTHER TASK FORCE                                  505
PATROL BORDER                                     477
OTHER AGENCY (TURNED OVER TO INS)                 444
WORKSITE ENFORCEMENT                              252
INSPECTIONS                                       127
ANTI-SMUGGLING                                     84
TRAFFIC CHECK                                      62
ORGANIZED CRIME DRUG ENFORCEMENT TASK FORCE        52
PATROL I

In [15]:
mk_key_columns=['apprehension_date','apprehension_state','birth_year','citizenship_country','gender','apprehension_site_landmark']

Potentially useful info in Final Program, but would have to do some more research to clarify meaning of some of the categories

## 2. Filtering - trying to find Mahmoud Khalil

### 1. Location and date:

#### In 26 Federal Plaza around 8th March 2025:
Note - taking a few days before and after 8th March because I don't know how immediately records are entered and processed.

In [21]:
arrests_df[filter_dates(arrests_df,'2025-03-07', end_date='2025-03-11') & loc_federal_plaza][mk_key_columns].sort_values(by='apprehension_date')

apprehension_date apprehension_state  birth_year  \
66791  2025-03-09 11:15:00           NEW YORK        1973   
181085 2025-03-10 08:49:00           NEW YORK        1985   
136002 2025-03-10 08:58:00           NEW YORK        1987   
90940  2025-03-10 10:27:00           NEW YORK        1988   
208815 2025-03-11 09:15:13           NEW YORK        1990   
120693 2025-03-11 10:00:00           NEW YORK        1999   
121873 2025-03-11 11:15:00                NaN        1990   
49911  2025-03-11 11:23:00           NEW YORK        1999   
15260  2025-03-11 13:29:00           NEW YORK        1987   

               citizenship_country  gender     apprehension_site_landmark  
66791                     HONDURAS  FEMALE  NDD - 26 FEDERAL PLAZA NY, NY  
181085                  BANGLADESH    MALE  NDD - 26 FEDERAL PLAZA NY, NY  
136002                     ECUADOR    MALE  NDD - 26 FEDERAL PLAZA NY, NY  
90940                      ECUADOR  FEMALE  NDD - 26 FEDERAL PLAZA NY, NY  
208815                  MONTENEGRO    MALE  NDD - 26 FEDERAL PLAZA NY, NY  
120693                     ECUADOR  FEMALE  NDD - 26 FEDERAL PLAZA NY, NY  
121873  CHINA, PEOPLES REPUBLIC OF    MALE  NDD - 26 FEDERAL PLAZA NY, NY  
49911                        INDIA    MALE  NDD - 26 FEDERAL PLAZA NY, NY  
15260                         PERU    MALE  NDD - 26 FEDERAL PLAZA NY, NY

#### In New York around 8th March 2025:

In [22]:
arrests_df[filter_dates(arrests_df,'2025-03-07', end_date='2025-03-10') & state_ny][mk_key_columns].sort_values(by='apprehension_date')

apprehension_date apprehension_state  birth_year citizenship_country  \
142348 2025-03-07 10:07:00           NEW YORK        1997           GUATEMALA   
17660  2025-03-07 10:12:28           NEW YORK        1970               BURMA   
172444 2025-03-07 10:54:00           NEW YORK        1977            HONDURAS   
116429 2025-03-07 10:59:00           NEW YORK        1995         EL SALVADOR   
178822 2025-03-08 00:23:00           NEW YORK        1995         EL SALVADOR   
162513 2025-03-08 09:18:00           NEW YORK        1995           VENEZUELA   
73250  2025-03-09 10:52:00           NEW YORK        1995             ECUADOR   
66791  2025-03-09 11:15:00           NEW YORK        1973            HONDURAS   
169123 2025-03-09 11:15:00           NEW YORK        1996            HONDURAS   
60359  2025-03-09 15:13:00           NEW YORK        1965            HONDURAS   
227318 2025-03-09 18:21:00           NEW YORK        2005             ECUADOR   
158918 2025-03-10 08:40:00           NEW YORK        1994             ECUADOR   
181085 2025-03-10 08:49:00           NEW YORK        1985          BANGLADESH   
136002 2025-03-10 08:58:00           NEW YORK        1987             ECUADOR   
161582 2025-03-10 09:23:00           NEW YORK        1985              RUSSIA   
171212 2025-03-10 09:46:00           NEW YORK        1999             ECUADOR   
90940  2025-03-10 10:27:00           NEW YORK        1988             ECUADOR   
35693  2025-03-10 10:46:00           NEW YORK        1996             ECUADOR   

        gender              apprehension_site_landmark  
142348  FEMALE                             ERIE COUNTY  
17660     MALE          MIDSTATE CORRECTIONAL FACILITY  
172444    MALE  CAP - USM - SOUTHERN DISTRICT NY STATE  
116429    MALE          CIP GENERAL AREA, NON-SPECIFIC  
178822    MALE          CIP GENERAL AREA, NON-SPECIFIC  
162513    MALE                           ALBANY COUNTY  
73250     MALE                           MONROE COUNTY  
66791   FEMALE           NDD - 26 FEDERAL PLAZA NY, NY  
169123  FEMALE            FUGITIVE OPERATIONS NY STATE  
60359     MALE            FUGITIVE OPERATIONS NY STATE  
227318    MALE            FUGITIVE OPERATIONS NY STATE  
158918    MALE          NBG GENERAL AREA, NON-SPECIFIC  
181085    MALE           NDD - 26 FEDERAL PLAZA NY, NY  
136002    MALE           NDD - 26 FEDERAL PLAZA NY, NY  
161582    MALE          NEW GENERAL AREA, NON-SPECIFIC  
171212    MALE                             ERIE COUNTY  
90940   FEMALE           NDD - 26 FEDERAL PLAZA NY, NY  
35693     MALE                             ERIE COUNTY

#### State missing around 8th March 2025:

In [37]:
arrests_df[filter_dates(arrests_df,'2025-03-08', end_date='2025-03-10') & state_missing][
    mk_key_columns].sort_values(by='apprehension_date')[mk_key_columns].iloc[50:100]

apprehension_date apprehension_state  birth_year  \
12323  2025-03-08 12:37:00                NaN        2000   
157165 2025-03-08 12:52:00                NaN        2002   
201250 2025-03-08 13:04:32                NaN        1999   
157786 2025-03-08 13:09:00                NaN        2004   
19623  2025-03-08 13:10:00                NaN        2001   
153911 2025-03-08 13:25:00                NaN        1984   
93328  2025-03-08 13:44:00                NaN        1996   
141546 2025-03-08 14:34:00                NaN        1980   
211740 2025-03-08 14:58:00                NaN        1995   
180013 2025-03-08 15:11:00                NaN        1999   
77948  2025-03-08 18:37:00                NaN        2002   
90799  2025-03-08 18:43:00                NaN        2000   
154443 2025-03-08 20:35:41                NaN        1977   
98054  2025-03-08 22:13:00                NaN        2001   
131875 2025-03-08 22:29:00                NaN        1992   
262017 2025-03-09 04:43:00                NaN        1992   
52308  2025-03-09 04:55:00                NaN        1999   
44430  2025-03-09 05:00:00                NaN        1992   
154545 2025-03-09 05:10:00                NaN        1979   
250929 2025-03-09 05:28:00                NaN        1982   
122169 2025-03-09 05:40:33                NaN        1987   
60537  2025-03-09 06:07:48                NaN        2005   
4505   2025-03-09 06:28:32                NaN        1977   
30262  2025-03-09 07:03:01                NaN        1981   
42620  2025-03-09 07:39:00                NaN        1988   
97102  2025-03-09 07:46:00                NaN        1990   
61307  2025-03-09 08:24:00                NaN        1985   
36294  2025-03-09 08:30:00                NaN        2002   
56503  2025-03-09 08:32:45                NaN        1988   
136230 2025-03-09 09:00:00                NaN        2001   
197863 2025-03-09 09:13:00                NaN        1987   
40301  2025-03-09 09:18:49                NaN        2006   
173133 2025-03-09 09:26:37                NaN        1991   
10039  2025-03-09 09:29:40                NaN        1985   
237469 2025-03-09 09:43:00                NaN        1973   
14436  2025-03-09 09:49:00                NaN        1984   
181361 2025-03-09 10:26:00                NaN        2000   
172921 2025-03-09 10:31:44                NaN        1975   
115133 2025-03-09 10:53:00                NaN        2002   
31117  2025-03-09 10:56:00                NaN        1977   
140140 2025-03-09 10:57:00                NaN        1991   
65095  2025-03-09 10:58:00                NaN        2002   
184131 2025-03-09 11:35:00                NaN        2004   
191838 2025-03-09 12:12:00                NaN        1989   
127903 2025-03-09 12:32:00                NaN        1996   
9014   2025-03-09 12:33:00                NaN        1983   
78885  2025-03-09 12:40:00                NaN        1992   
138427 2025-03-09 12:40:37                NaN        1997   
207992 2025-03-09 12:51:00                NaN        1978   
169345 2025-03-09 13:45:00                NaN        1990   

               citizenship_country  gender  \
12323                    GUATEMALA    MALE   
157165                    CAMEROON  FEMALE   
201250                   VENEZUELA    MALE   
157786                    HONDURAS    MALE   
19623                       MEXICO    MALE   
153911                   GUATEMALA    MALE   
93328                     HONDURAS    MALE   
141546                      MEXICO    MALE   
211740                   GUATEMALA    MALE   
180013                    COLOMBIA    MALE   
77948                     HONDURAS    MALE   
90799                       MEXICO    MALE   
154443  CHINA, PEOPLES REPUBLIC OF  FEMALE   
98054                         CUBA    MALE   
131875                      MEXICO    MALE   
262017                      MEXICO    MALE   
52308                  EL SALVADOR    MALE   
44430                     HONDURAS    MALE   
154545                 

**Summary:** There does not seem to be a clear match for either location field around the timeframe. 

### 2. Demographics

#### Citizenship == Algeria, and Birth Year is 1995:

In [35]:
arrests_df[citizenship_algeria & birth_1995][mk_key_columns]

apprehension_date apprehension_state  birth_year citizenship_country  \
107560 2025-03-11 12:56:29           NEW YORK        1995             ALGERIA   

       gender       apprehension_site_landmark  
107560   MALE  SING SING CORRECTIONAL FACILITY

### Could this be Mahmoud Khalil?

**Headline:** I think it is unlikely. While there are some clear similarities, there are important inconsistancies that don't fit what has been stated in the media, and by Mahmoud Khalil himself, about the case.

Evidence in support:
* Apprehension in New York close to the time Mahmoud Khalil was arrested
* Demographic match - Citizenship is Algerian, and Birth Year is 1995

Evidence against:
* Apprehension site landmark is Sing Sing Correctional Facility, which is a prison in upstate New York. There is no reporting or other evidence to suggest Mahmoud Khalil was taken to Sing Sing, and the timeframe does not really allow for it (he was stated to be in Louisiana around the same time)
* Apprehension method is "CAP STATE INCARCERATION" - CAP is Criminal Alien Program ([source: ICE.gov](https://www.ice.gov/identify-and-arrest/criminal-alien-program)), and "CAP STATE INCARCERATION" suggests this is an example of someone already incarcerated being arrested by ICE 
